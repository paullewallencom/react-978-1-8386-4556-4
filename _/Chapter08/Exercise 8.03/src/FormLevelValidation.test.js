import React from 'react';
import {render, fireEvent, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import FormLevelValidation from './FormLevelValidation';

test('renders correctly', () => {
  const {container} = render(<FormLevelValidation/>);
  expect(container).toMatchSnapshot();
});

test('loads and displays form', async () => {
  //Arrange
  const { container } = render(<FormLevelValidation />);
  const email = container.querySelector('input[type="text"]');
  const password = container.querySelector('input[type="password"]');

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: 'example@email.com'
      }
    });
    fireEvent.change(password, {
      target: {
        value: 'password'
      }
    })
  });

  //Assert
  expect(email.value).toBe('example@email.com');
  expect(password.value).toBe('password');
});


test('displays an error message', async () => {
  //Arrange
  const { container } = render(<FormLevelValidation />);
  const email = container.querySelector('input[type="text"]');
  const password = container.querySelector('input[type="password"]');

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: ''
      }
    });
    fireEvent.change(password, {
      target: {
        value: ''
      }
    });
    fireEvent.blur(password);
    fireEvent.blur(email);
  });


  //Assert
  expect(container).toMatchSnapshot();
});


