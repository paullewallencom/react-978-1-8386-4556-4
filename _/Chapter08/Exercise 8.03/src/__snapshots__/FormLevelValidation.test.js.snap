// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an error message 1`] = `
<div>
  <form>
    <label>
      Name*:
      <input
        name="name"
        type="text"
        value=""
      />
    </label>
    <span
      class="error"
    >
      Name is Required
    </span>
    <label>
      Password*:
      <input
        name="password"
        type="password"
        value=""
      />
    </label>
    <span
      class="error"
    >
      Password is Required
    </span>
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <form>
    <label>
      Name*:
      <input
        name="name"
        type="text"
        value=""
      />
    </label>
    <label>
      Password*:
      <input
        name="password"
        type="password"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
